tds.go

```diff
@@ -995,43 +1175,64 @@ initiate_connection:
 	}
 
 	// processing login response
-	reader := startReading(&sess, ctx, nil)
-	for {
-		tok, err := reader.nextToken()
-		if err == nil {
-			if tok == nil {
-				break
-			} else {
-				switch token := tok.(type) {
-				case sspiMsg:
-					sspi_msg, err := auth.NextBytes(token)
-					if err != nil {
-						return nil, err
-					}
-					if len(sspi_msg) > 0 {
-						outbuf.BeginPacket(packSSPIMessage, false)
-						_, err = outbuf.Write(sspi_msg)
+
+	for pending := true; pending; {
```

Comment:
If you need to break out of a loop in Go, label it and then say "break myLabel".

Reply:
The outer loop needs to break only at the conclusion of the inner loop, though - so I set the flag in the inner loop and then test in the outer loop so it breaks at the right time. I could make that outer loop infinite and use an if pending { break }, but I think that's not what you're asking.